Assets/MixedRealityToolkit/Providers/BaseSpatialAwarenessObserver.cs

```diff
@@ -96,7 +95,7 @@ public override int GetHashCode()
 
         private GameObject observedObjectParent = null;
 
-        public GameObject ObservedObjectParent => observedObjectParent ?? (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent(Name));
+        public GameObject ObservedObjectParent => observedObjectParent != null ? ObservedObjectParent : (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent(Name));
```

Comment:
This looks like it'll overflow the stack by calling `ObservedObjectParent` repeatedly.

```suggestion
        public GameObject ObservedObjectParent => observedObjectParent != null ? observedObjectParent : (observedObjectParent = SpatialAwarenessSystem?.CreateSpatialAwarenessObjectParent(Name));
```